fluss-flink/fluss-flink-common/src/main/java/com/alibaba/fluss/flink/tiering/source/enumerator/TieringSourceEnumerator.java

```diff
@@ -0,0 +1,470 @@
+                    waitHeartbeatResponse(
+                            coordinatorGateway.lakeTieringHeartbeat(
+                                    heartBeatWithRequestNewTieringTable(tieringHeartbeatRequest)));
+            if (heartbeatResponse.hasTieringTable()) {
+                PbLakeTieringTableInfo tieringTable = heartbeatResponse.getTieringTable();
+                return Tuple3.of(
+                        tieringTable.getTableId(),
+                        tieringTable.getTieringEpoch(),
+                        TablePath.of(
+                                tieringTable.getTablePath().getDatabaseName(),
+                                tieringTable.getTablePath().getTableName()));
+            } else {
+                LOG.info("No available Tiering table found, will poll later.");
+            }
+        } else {
+            // report heartbeat to fluss coordinator
+            waitHeartbeatResponse(coordinatorGateway.lakeTieringHeartbeat(tieringHeartbeatRequest));
+        }
+        return null;
+    }
+
+    private void generateTieringSplits(Tuple3<Long, Long, TablePath> tieringTable)
+            throws FlinkRuntimeException {
+        if (tieringTable == null) {
+            return;
+        }
+        long start = System.currentTimeMillis();
+        LOG.info("Generate Tiering splits for table {}.", tieringTable.f2);
+        try {
+            List<TieringSplit> tieringSplits = splitGenerator.generateTableSplits(tieringTable.f2);
+            LOG.info(
+                    "Generate Tiering {} splits for table {} with cost {}ms.",
+                    tieringSplits.size(),
+                    tieringTable.f2,
+                    System.currentTimeMillis() - start);
+            if (tieringSplits.isEmpty()) {
+                LOG.info(
+                        "Generate Tiering splits for table {} is empty, no need to tier data.",
+                        tieringTable.f2.getTableName());
+                finishedTableEpochs.put(tieringTable.f0, tieringTable.f1);
+            } else {
+                tieringTableEpochs.put(tieringTable.f0, tieringTable.f1);
+                pendingSplits.addAll(tieringSplits);
+            }
+        } catch (Exception e) {
+            throw new FlinkRuntimeException(
+                    String.format(
+                            "Generate Tiering splits for table %s failed due to:", tieringTable.f1),
+                    e);
+        }
+    }
+
+    @Override
+    public TieringSourceEnumeratorState snapshotState(long checkpointId) throws Exception {
+        // do nothing, the downstream lake commiter will snapshot the state to Fluss Cluster
+        return new TieringSourceEnumeratorState();
+    }
+
+    @Override
+    public void close() throws IOException {
+        if (rpcClient != null) {
+            failedTableEpochs.putAll(tieringTableEpochs);
+            tieringTableEpochs.clear();
+            reportFailedTable(basicHeartBeat(), failedTableEpochs);
```

Comment:
nit:
```
if (!failedTableEpochs.isEmpty()) {
    reportFailedTable(basicHeartBeat(), failedTableEpochs);
}
```

Reply:
When close this enumerator, the tieringTable under `tiering` without received `finished` signal  should be treated as `failed` ?